### PR TITLE
Update safe area calculations

### PR DESCRIPTION
**Pull request by tcobbs-bentley**
Updates are based on code from SYNCHRO Field that was designed to work in MultiWindowMode.

### CODE REVIEW

File: mobile-sdk/src/main/java/com/github/itwin/mobilesdk/ITMApplication.kt [hunk 1]
```diff
@@ -592,38 +594,56 @@ open class ITMApplication(
...
         }
+        var insetsType = WindowInsetsCompat.Type.displayCutout()
+        // Ideally we'd always get system bars insets but before API 30 getInsets() returns bars insets even if they are overlaid on top.
+        if (activity.isInMultiWindowMode)
+            insetsType = insetsType or WindowInsetsCompat.Type.systemBars()
+
+        val safeAreaInsets = WindowInsetsCompat.toWindowInsetsCompat(insets, view).getInsets(insetsType)
+        return safeAreaInsets.let {
+            Quint(toDp(max(it.left, it.right)), toDp(it.left), toDp(it.right), toDp(it.top), toDp(it.bottom))
+        }
```
Comment (toddsouthenbentley) [inline mobile-sdk/src/main/java/com/github/itwin/mobilesdk/ITMApplication.kt:630]: I know the code came from Field but the use of `let` here seems odd and unnecessary to me. It seems like they used `let` just to make the parameters to the `Quint` constructor shorter. The same could be accomplished by renaming `safeAreaInsets` to something shorter:

```suggestion
        val in = WindowInsetsCompat.toWindowInsetsCompat(insets, view).getInsets(insetsType)
        return Quint(toDp(max(in.left, in.right)), toDp(in.left), toDp(in.right), toDp(in.top), toDp(in.bottom))
```

Comment (toddsouthenbentley) [reply mobile-sdk/src/main/java/com/github/itwin/mobilesdk/ITMApplication.kt:630]: Another possible option:

```suggestion
        return WindowInsetsCompat.toWindowInsetsCompat(insets, view).getInsets(insetsType).let {
            Quint(toDp(max(it.left, it.right)), toDp(it.left), toDp(it.right), toDp(it.top), toDp(it.bottom))
        }
```

Comment (tcobbs-bentley) [reply mobile-sdk/src/main/java/com/github/itwin/mobilesdk/ITMApplication.kt:630]: I'm going to instead get rid of the variable entirely and use `with()`.

Comment (tcobbs-bentley) [reply mobile-sdk/src/main/java/com/github/itwin/mobilesdk/ITMApplication.kt]: Done: as an aside, the Quint was all me. The SYNCHRO code only used sides, top, and bottom, so they used a (built-in) Triple. I created the Quint class to add left and right to the return value to support the minLeft and minRight fields that they didn't include.

**Review by toddsouthenbentley — APPROVED** [review]

**Pull request merged**